src/cluster.c

```diff
@@ -7005,15 +7005,25 @@ void slotToKeyDestroy(redisDb *db) {
  * The number of removed items is returned. */
 unsigned int delKeysInSlot(unsigned int hashslot) {
     unsigned int j = 0;
+    server.propagate_no_multi = 1;
     dictEntry *de = (*server.db->slots_to_keys).by_slot[hashslot].head;
     while (de != NULL) {
         sds sdskey = dictGetKey(de);
         de = dictEntryNextInSlot(de);
         robj *key = createStringObject(sdskey, sdslen(sdskey));
         dbDelete(&server.db[0], key);
+        signalModifiedKey(NULL, &server.db[0], key);
+        moduleNotifyKeyspaceEvent(NOTIFY_GENERIC, "del", key, server.db[0].id);
```

Comment:
considering the fresh topic of #10969, we should put the key-space notification after the deletion and propagation..

Reply:
i'm not sure.
maybe after we reverted the changes in 10969 we concluded the only thing that matters is that it's after the actual deletion, and the order with propagation doesn't matter (since KSN should only do read-only actions)?
i can't seem to be able to find any written message about that.
@weim0000 do you recall anything?

Reply:
and https://github.com/redis/redis/pull/11084#issuecomment-1209433976

> I agree, sending keyspace notification about deletions will cause a similar damage as i mentioned above (the keys were not really deleted from the database, just moved), but i do think modules need to know these keys are no longer locally available, so i think we should send some keyspace notification just to modules.

Reply:
these comments just say that it may be wrong to issue keyspace notification to clients, but we do want to send a notification to modules.

the question @MeirShpilraien asks is why we didn't put the call to `moduleNotifyKeyspaceEvent` before `propagatePendingCommands`.
i guess we just forgot.

anyway, Meir is fixing that in #11199